geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/serialization/TwkbSerialization.scala

```diff
@@ -541,12 +559,19 @@ trait TwkbSerialization[T <: NumericWriter, V <: NumericReader]
 
   private class XYMState(precision: Int, mPrecision: Int) extends ExtendedState(precision) {
 
-    private val pm: Double = math.pow(10, mPrecision)
-    private var m: Int = 0
+    protected val pm: Double = math.pow(10, mPrecision)
+    protected var m: Int = 0
 
     // sets bit for m dim, and its precisions
     override protected val extendedDims: Byte = (0x02 | ((mPrecision & 0x03) << 5)).toByte
 
+    override protected def readCoordinateState(in: V): Unit = {
+      super.readCoordinateState(in)
+      m = m + readVarInt(in)
```

Comment:
The small part of me that is interested in over-optimizing things wonders if the various function dispatches here would add up to anything worth discussing.

The part of me that's a Scala fan wants to think through linearization so that the class structure could mix-in the z and m handling in separate traits....